### PR TITLE
Added command to upgrade CLI to the latest version(#62)

### LINKED ISSUE #62: In-place upgrade for CLI
Command: `spice upgrade`

This will self-upgrade the CLI to the latest version. Once the CLi is updated, when it's run, the CLI will automatically upgrade to the latest runtime version.

Would be a new CLI command which would manually call: `context.InstallOrUpgradeRuntime()` - See https://github.com/spiceai/spiceai/blob/94427e2ba0257af06d008f5143a9641ac3ddfa78/pkg/context/context.go#L18

### PR DESCRIPTION
**Pull request by Adm28**
Added `spice upgrade` command which upgrades the CLI to the latest version.

`> spice upgrade`

```
Checking for latest Spice CLI release...
Upgrading the Spice.ai CLI ...
Spice.ai CLI upgraded to v0.3-alpha successfully.
```

### CODE REVIEW

File: pkg/cli/cmd/upgrade.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		release, err := github.GetLatestCliRelease()
+		if err != nil {
+			cmd.PrintErrln("Error checking for latest release: %w", err)
+			return
+		}
+
+		rtcontext := context.CurrentContext()
+		cliVersion := version.Version()
+
+		if semver.Compare(cliVersion, release.TagName) == 0 {
```
Comment (phillipleblanc) [inline pkg/cli/cmd/upgrade.go]: This should also compare by trimming the `-alpha` suffix

Comment (lukekim) [inline pkg/cli/cmd/upgrade.go]:
```suggestion
		if cliVersion == release.TagName {
```

File: pkg/github/runtime_release.go [hunk 2]
```diff
@@ -17,10 +17,21 @@ const (
 	runtimeRepo  = "spiceai"
 )
 
-func GetLatestRuntimeRelease(tagName string) (*RepoRelease, error) {
+func GetLatestRuntimeRelease() (*RepoRelease, error) {
 	fmt.Println("Checking for latest Spice runtime release...")
 
-	release, err := GetLatestRelease(githubClient, tagName, GetRuntimeAssetName())
+	release, err := GetLatestRelease(githubClient, GetAssetName(constants.SpiceRuntimeFilename))
```
Comment (lukekim) [inline pkg/github/runtime_release.go:23]: this can just

```go
return GetLatestRelease(
```

File: pkg/cli/cmd/upgrade.go (hunk 1 above)
Comment (phillipleblanc) [reply pkg/cli/cmd/upgrade.go]: Actually, it probably makes more sense to just do a straight string comparison

Comment (Adm28) [reply pkg/cli/cmd/upgrade.go]: @phillipleblanc Should I replace the compare function with straight string comparison ?

Comment (phillipleblanc) [reply pkg/cli/cmd/upgrade.go]: Yeah, I think that makes the most sense here.

**Review by lukekim — APPROVED** [review]

**Pull request merged**